frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx

```diff
@@ -0,0 +1,205 @@
+import * as classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import * as React from 'react';
+import { ActionGroup, Button, FormGroup, Form, TextInput, Tooltip } from '@patternfly/react-core';
+import {
+  ButtonBar,
+  Dropdown,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { apiVersionForModel, k8sCreate } from '@console/internal/module/k8s';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { SecretModel } from '@console/internal/models';
+import { CEPH_STORAGE_NAMESPACE } from '@console/ceph-storage-plugin/src/constants';
+import { initialState, providerDataReducer } from './reducer';
+import { NooBaaNamespaceStoreModel } from '../../models';
+import {
+  BC_PROVIDERS,
+  NOOBAA_TYPE_MAP,
+  PROVIDERS_NOOBAA_MAP,
+  BUCKET_LABEL_NOOBAA_MAP,
+} from '../../constants';
+import { getExternalProviders, getProviders, secretPayloadCreator } from '../../utils/noobaa-utils';
+import { Payload } from '../../types';
+import '../endpoints/endpoints.scss';
+import { S3EndPointType } from '../endpoints/s3-endpoint-type';
+import { StoreType } from '../../constants/common';
+
+const PROVIDERS = getProviders(StoreType.NS);
+const externalProviders = getExternalProviders(StoreType.NS);
+
+const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
+  NamespaceStoreFormProps & HandlePromiseProps
+>((props) => {
+  const { t } = useTranslation();
+  const [nsName, setNsName] = React.useState('');
+  const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
+  const [providerDataState, providerDataDispatch] = React.useReducer(
+    providerDataReducer,
+    initialState,
+  );
+
+  const handleNsNameTextInputChange = (strVal: string) => {
+    if (strVal.length <= 43) {
+      setNsName(strVal);
+    }
+  };
```

Comment:
I see you are setting `maxLength` then this function wont be called after maxLength
```suggestion
  const handleNsNameTextInputChange = (strVal: string) =>  setNsName(strVal);

```

Reply:
makes sense. Changed.